tuned/profiles/functions/function_calc_isolated_cores.py

```diff
@@ -0,0 +1,44 @@
+import os
+import glob
+import tuned.logs
+from . import base
+import tuned.consts as consts
+
+log = tuned.logs.get()
+
+class calc_isolated_cores(base.Function):
+	"""
+	Calculates and returns isolated cores. The argument specifies how many
+	cores per socket reserve for housekeeping. If not specified, 1 core
+	per socket is reserved for housekeeping and the rest is isolated.
+	"""
+	def __init__(self):
+		# max 1 argument
+		super(calc_isolated_cores, self).__init__("calc_isolated_cores", 1)
+
+	def execute(self, args):
+		if not super(calc_isolated_cores, self).execute(args):
+			return None
+		cpus_reserve = 1
+		if len(args) > 0:
+			if not args[0].isdecimal() or int(args[0]) < 0:
```

Comment:
Shouldn't there be `int(args[0]) <= 0` to always have at least one housekeeping core? If there can be no isolated core the error log on next line probably should say `... non-negative integer`

Reply:
0 is valid setting - to have no housekeeping core. I will update the error message.